### ISSUE TITLE
Detection of s3 download does not respect files on xx.s3.amazonaws.com

### ISSUE THREAD

**Issue by moorscode** [easydigitaldownloads/edd-amazon-s3#28]
We are using a plugin to upload our content to S3 directly.

This means that our downloads have the following format:
`https://our-bucket.s3.amazonaws.com/uploads/edd/2015/09/{filename}`

We are using a different bucket for these uploads as the default edd-amazon-s3.
These files are not detected as "is_s3_download" and will be formatted again to an unusable URL.

PR is coming up to provide the detection of files already present in an s3 bucket to apply secret and timeout.

**Comment by pippinsplugins** [comment]
@moorscode Thanks for the report!

To help better understand the exact issue, can you elaborate a little bit more on why these files don't work? Are you inserting them into the EDD upload field with the complete URL to the file in the S3 bucket?

**Comment by moorscode** [comment]
Yes, we use a plugin that uploads the images to a specific bucket and adds the complete URL in the field to use.
Thus we have links to a different bucket than the bucket configured in the s3 plugin.

**Comment by pippinsplugins** [comment]
Thanks. It would be nice to account for those.

I see a few things that will need tweaked in the PR. See my comment here? https://github.com/easydigitaldownloads/edd-amazon-s3/pull/29/files#r63301600

**Comment by moorscode** [comment]
Yes, absolutely agree, will improve on this!

**Comment by pippinsplugins** [comment]
:+1:

**Comment by moorscode** [comment]
@pippinsplugins I've added a lot of commentary and split up the code a bit.
If you need me to change anything else please let me know :)

**Comment by pippinsplugins** [comment]
Apologies for the delay on this. I'm going to try and review / test it further later this week.

**Comment by pippinsplugins** [comment]
@moorscode any chance you could update your PR to merge cleanly?
